src/Avalonia.Visuals/Media/Immutable/ImmutableSolidColorBrush.cs

```diff
@@ -55,5 +55,32 @@ public override string ToString()
         {
             return Color.ToString();
         }
+
+        /// <inheritdoc/>
+        public bool Equals(ISolidColorBrush other)
+        {
+            return other != null
+                ? Hash(this) == Hash(other)
```

Comment:
I don't think comparing hashes is the right thing to do here. Hashes can collide (because you're compressing 4 `byte`s and a `double` into an `int`) but `Equals` should produce the right result even in the presence of a hash collision.

Reply:
is resolved